File: TeddyPet_BE/src/main/java/fpt/teddypet/application/dto/request/auth/LoginRequest.java
package fpt.teddypet.application.dto.request.auth;
import jakarta.validation.constraints.NotBlank;

public record LoginRequest(
        @NotBlank(message = "Username or Email is required")
        String usernameOrEmail,
        
        @NotBlank(message = "Password is required")
        String password
) {
}

File: teddypet_mobile/lib/data/models/request/auth/login_request.dart
// lib/data/models/request/auth/login_request.dart
class LoginRequest {
  final String usernameOrEmail;
  final String password;

  LoginRequest({
    required this.usernameOrEmail,
    required this.password,
  });

  Map<String, dynamic> toJson() {
    return {
      'usernameOrEmail': usernameOrEmail,
      'password': password,
    };
  }
}
